packages/sdk/src/decode/decode.ts

```diff
@@ -40,40 +130,61 @@ const decodeArray = async (array, decoders, stack) =>
     )
   )
 
-const decodeDictionary = async (dictionary, decoders, stack) =>
-  await dictionary.reduce(async (acc, v) => {
-    acc = await acc
-    acc[await recurseDecode(v.key, decoders, [...stack, v.key])] =
-      await recurseDecode(v.value, decoders, [...stack, v.key])
-    return acc
-  }, Promise.resolve({}))
-
-const decodeComposite = async (composite, decoders, stack) => {
-  const decoded = await composite.fields.reduce(async (acc, v) => {
-    acc = await acc
-    acc[v.name] = await recurseDecode(v.value, decoders, [...stack, v.name])
-    return acc
-  }, Promise.resolve({}))
+const decodeDictionary = async (
+  dictionary: KeyValuePair[],
+  decoders: DecoderMap,
+  stack: any[]
+): Promise<Record<string, any>> =>
+  await dictionary.reduce(
+    async (acc, v) => {
+      acc = await acc
+      acc[await recurseDecode(v.key, decoders, [...stack, v.key])] =
+        await recurseDecode(v.value, decoders, [...stack, v.key])
+      return acc
+    },
+    Promise.resolve({}) as any
+  )
+
+const decodeComposite = async (
+  composite: CompositeInstruction,
+  decoders: DecoderMap,
+  stack: any[]
+): Promise<any> => {
+  const decoded = await composite.fields.reduce(
+    async (acc, v) => {
+      acc = await acc
+      acc[v.name] = await recurseDecode(v.value, decoders, [...stack, v.name])
+      return acc
+    },
+    Promise.resolve({}) as any
+  )
   const decoder = composite.id && decoderLookup(decoders, composite.id)
   return decoder ? await decoder(decoded) : decoded
 }
 
-const decodeInclusiveRange = async (range, decoders, stack) => {
+const decodeInclusiveRange = async (
+  range: InclusiveRangeValue,
+  decoders: DecoderMap,
+  stack: any[]
+): Promise<Record<string, any>> => {
   // Recursive decode for start, end, and step
   // We don't do all fields just in case there are future API changes
   // where fields added and are not Cadence values
   const keys = ["start", "end", "step"]
-  const decoded = await Object.keys(range).reduce(async (acc, key) => {
-    acc = await acc
-    if (keys.includes(key)) {
-      acc[key] = await recurseDecode(range[key], decoders, [...stack, key])
-    }
-    return acc
-  }, Promise.resolve({}))
+  const decoded = await Object.keys(range).reduce(
```

Comment:
Not that I want to add something that could break or needs testing. Maybe in a separate PR, we could improve the performance of this function to use `Promise.all` instead of sequentially calling these.

Reply:
Yes, that makes sense. There were some improvements that could have been made, but I intentionally left them out to focus only on the TypeScript conversion. This was to avoid introducing potential issues or bugs that could have made the review process more complex.